dir.targets

```diff
@@ -70,4 +70,24 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="RewriteReferences" Condition="'$(SourceIndex)' == 'true'" AfterTargets="ResolveNuGetPackages">
+    <ItemGroup>
+      <Reference Remove="@(_ReferenceFromPackage)"/>
+      <_ReferenceProjectToHarvest Include="@(_ReferenceFromPackage -> '$(MSBuildThisFileDirectory)src/%(Filename)/src/%(FileName).builds')" />
+    </ItemGroup>
+
+    <MSBuild Condition="'$(SourceIndex)' == 'true' AND Exists('%(Identity)')"
+             Projects="%(_ReferenceProjectToHarvest.Identity)"
+             Targets="GetTargetPath">
+      <Output TaskParameter="TargetOutputs" ItemName="_PackageLibReferencePath"/>
+    </MSBuild>
+    <Error Condition="!Exists('%(_PackageLibReferencePath.Identity)')" Text="Required Reference '%(Identity)' does not exist. Ensure that you have built the entire tree."/>
+    <ItemGroup Condition=" '$(SourceIndex)' == 'true' ">
+      <ReferencePath Include="%(_PackageLibReferencePath.Identity)" Condition="Exists('%(Identity)')"/>
+      <ReferencePath Include="$(PackagesDir)runtime.win7-x64.Microsoft.NETCore.Runtime.CoreCLR\1.2.0-$(CoreClrExpectedPrerelease)\runtimes\win7-x64\lib\netstandard1.0\mscorlib.dll"/>
```

Comment:
We have the same problem https://github.com/dotnet/corefx/blob/master/layout/dir.proj#L20 and it does keep breaking. We really should figure out a better way to handle some of these things. We can probably add the entire version in dependencies.props file and at a minimum check for the existence of that file and even probably update it automatically as part of our update PR's.